coderd/rbac/roles.go

```diff
@@ -37,39 +38,96 @@ func init() {
 // RoleNames is a list of user assignable role names. The role names must be
 // in the builtInRoles map. Any non-user assignable roles will generate an
 // error on Expand.
-type RoleNames []string
+type RoleNames []RoleName
 
 func (names RoleNames) Expand() ([]Role, error) {
 	return rolesByNames(names)
 }
 
-func (names RoleNames) Names() []string {
+func (names RoleNames) Names() []RoleName {
 	return names
 }
 
-// The functions below ONLY need to exist for roles that are "defaulted" in some way.
-// Any other roles (like auditor), can be listed and let the user select/assigned.
-// Once we have a database implementation, the "default" roles can be defined on the
-// site and orgs, and these functions can be removed.
+// RoleName contains both the name of the role, and any organizational scope.
+// Both fields are required to be globally unique and identifiable.
+type RoleName struct {
+	Name string
+	// OrganizationID is uuid.Nil for unscoped roles (aka deployment wide)
+	OrganizationID uuid.UUID
+}
```

Comment:
we should probably call this something else -- maybe `RoleIdentifier`? chef's choice